### ISSUE TITLE
[METEOR 3] error on boot.js TypeError: Meteor.methods is not a function

### ISSUE THREAD

**Issue by StorytellerCZ** [meteor/meteor#12943]
When starting Meteor 3.0-beta.0 with packages that call `Meteor.methods` the following error prevents startup:

```
I20240101-17:45:42.416(1)? error on boot.js TypeError: Meteor.methods is not a function
I20240101-17:45:42.416(1)?     at module.wrapAsync.self (packages/socialize:server-time/server/server-time.js:10:8)
I20240101-17:45:42.416(1)?     at Module.wrapAsync (/home/storyteller/.meteor/packages/modules/.0.19.1-beta300.0.6m957j.xpa8p++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/@meteorjs/reify/lib/runtime/index.js:251:8)
I20240101-17:45:42.416(1)?     at module (/home/storyteller/Web/meteor/3.0-beta.0/.meteor/local/build/programs/server/packages/socialize_server-time.js:18:9)
I20240101-17:45:42.416(1)?     at fileEvaluate (packages/modules-runtime.js:335:7)
I20240101-17:45:42.416(1)?     at Module.require (packages/modules-runtime.js:237:14)
I20240101-17:45:42.416(1)?     at Module.mod.require (/home/storyteller/.meteor/packages/modules/.0.19.1-beta300.0.6m957j.xpa8p++os+web.browser+web.browser.legacy+web.cordova/npm/node_modules/@meteorjs/reify/lib/runtime/index.js:30:33)
I20240101-17:45:42.416(1)?     at Object.require (packages/modules-runtime.js:257:21)
I20240101-17:45:42.416(1)?     at evaluateNextModule (packages/core-runtime.js:183:26)
I20240101-17:45:42.416(1)?     at runEagerModules (packages/core-runtime.js:222:3)
I20240101-17:45:42.416(1)?     at load (packages/core-runtime.js:141:3)
I20240101-17:45:42.416(1)?     at Object.queue (packages/core-runtime.js:134:5)
I20240101-17:45:42.416(1)?     at /home/storyteller/Web/meteor/3.0-beta.0/.meteor/local/build/programs/server/packages/socialize_server-time.js:1:47
I20240101-17:45:42.416(1)?     at /tools/static-assets/server/boot.js:432:36
I20240101-17:45:42.416(1)?     at /tools/static-assets/server/boot.js:503:13
I20240101-17:45:42.417(1)?     at AsyncLocalStorage.run (node:async_hooks:346:14)
I20240101-17:45:42.417(1)?     at /tools/static-assets/server/boot.js:502:37
```

**Replication**
* Create a new Meteor 3.0-beta.0 app (I used the React skeleton)
* Create `packages` folder, copy over a packages that calls `Meteor.methods`, for example `socialize:server-time` [GitHub](https://github.com/copleykj/socialize-server-time)
* Observe crash
* If you add `console.dir(Meteor)` into `/packages/socialize-server-time/server/server-time.js` it will show you that the `methods` functions and many other are missing from the `Meteor` object.
* If you print out Meteor in the app server it has all the necessary methods

My conclusion is that the methods and other (like collections) are add added after Meteor runs through boot of packages.
Could be potentially something with the load order? So potentially ddp-server might be loaded after third-party package?

**Comment by zodern** [comment]
If you add `api.use('ddp')` it should work. This issue is probably because Meteor 3 loads packages in parallel when possible: https://github.com/meteor/meteor/blob/release-3.0/docs/source/api/top-level-await.md#module-and-package-execution-order.

`socialize:server-time` requires `ddp-server` to load first. Even though `socialize:server-time` doesn't specify that in its package.js file, most apps would list it in `.meteor/packages` after `ddp-server` or after another package that uses `ddp-server` so it usually works in Meteor 2. Though, it would fail with the same error if it was listed earlier in `.meteor/packages`.

Since Meteor 3 uses the dependencies between packages to know when packages can load in parallel, it is more important for packages to list their dependencies instead of relying on the order of packages in `.meteor/packages`.

We chose to aggressively parallelize loading packages for Meteor 3 since it could speed up starting the server, and if we made the wrong choice it is easier to remove parallelism than to add it without breaking things. So far the only issue we've found is with packages that use `ddp-server` and don't specify that in their `package.js` file. I'm not sure if that should be a considered a bug in Meteor 3, or a bug with the package since there are ways the error could be encountered in Meteor 2.
